sips/instance_decided_enforcement.md

```diff
@@ -0,0 +1,152 @@
+```go
+// StartNewInstance will start a new QBFT instance, if can't will return error
+func (c *Controller) StartNewInstance(height Height, value []byte) error {
+	if err := c.GetConfig().GetValueCheckF()(value); err != nil {
+		return errors.Wrap(err, "value invalid")
+	}
+	
+	// can't use <= because of height == 0 case
+	if height < c.Height {
+		return errors.New("invalid instance height")
+	}
+	
+	// covers height == 0 case
+	if c.StoredInstances.FindInstance(height) != nil {
+		return errors.New("instance already running")
+	}
+	
+	c.Height = height
+
+	newInstance := c.addAndStoreNewInstance()
+	newInstance.Start(value, height)
+
+	
+	c.forceStopAllInstanceExceptCurrent()
+
+	return nil
+}
+
+func (c *Controller) forceStopAllInstanceExceptCurrent() {
+    for _, i := range c.StoredInstances {
+        if i.State.Height != c.Height {
+            i.ForceStop()
+        }
+    }
+}
+
+```
+
+**QBFT Instance**
+```go
+type Instance struct {
+	// ...
+    // forceStop will force stop the instance if set to true
+    forceStop  bool
+}
+
+func (i *Instance) ForceStop() {
+    i.forceStop = true
+}
+
+
+// CanProcessMessages will return true if instance can process messages
+func (i *Instance) CanProcessMessages() bool {
+	return i.State.Round < CutoffRound && !i.forceStop
+}
+```
+```go
+// ProcessMsg processes a new QBFT msg, returns non nil error on msg processing error
+func (i *Instance) ProcessMsg(msg *SignedMessage) (decided bool, decidedValue []byte, aggregatedCommit *SignedMessage, err error) {
+    if i.CanProcessMessages() {
+        return false, nil, nil, errors.New("instance stopped processing messages")
+    }
+	
+	// ...
+}
+```
+```go
+func (i *Instance) UponRoundTimeout() error {
+	if i.CanProcessMessages() {
+		return errors.New("instance stopped processing timeouts")
+	}
+
+	newRound := i.State.Round + 1
+	defer func() {
+		i.State.Round = newRound
+		i.State.ProposalAcceptedForCurrentRound = nil
+		i.config.GetTimer().TimeoutForRound(i.State.Round)
+	}()
+
+	roundChange, err := CreateRoundChange(i.State, i.config, newRound, i.StartValue)
+	if err != nil {
+		return errors.Wrap(err, "could not generate round change msg")
+	}
+
+	if err := i.Broadcast(roundChange); err != nil {
+		return errors.Wrap(err, "failed to broadcast round change message")
+	}
+
+	return nil
+}
+```
+
+
+**Runner**
+- Remove function canStartNewDuty
+- Start new instance with the duty's slot
+```go
+// ProcessMsg processes a new QBFT msg, returns non nil error on msg processing error
+if err := runner.GetBaseRunner().QBFTController.StartNewInstance(
+    qbft.Height(input.Duty.Slot),
+    byts,
+); err != nil {
+    return errors.Wrap(err, "could not start new QBFT instance")
+}
+```
+
+**Syncing**  
+- Light nodes do not need to sync highest decided at all
+- Syncing for decided messages is done by epoch range
```

Comment:
So light nodes still need to sync decided messages?

Reply:
I think @alonmuroch meant for full nodes?

Let's move discussion about full nodes syncing to https://github.com/bloxapp/ssv-spec/issues/280